Rubberduck.Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -97,9 +101,9 @@ private void RefactorImpl(Declaration inputTarget, IRenamePresenter presenter)
 
                 if (TrySetNewName(presenter))
                 {
-                    Rename();
-                    Rewrite();
-                    Reparse();
+                    var rewriteSession = _rewritingManager.CheckOutCodePaneSession();
+                    Rename(rewriteSession);
+                    rewriteSession.TryRewrite();
```

Comment:
I'm a bit annoyed that each Refactoring implementation needs to remember doing that. Is there a clean way to fix this?

I assume there's some issues with refactorings that use dialogs that makes it hard to correctly unify this...

Reply:
They all require a session to be able to rewrite and that session and the changes need to be applied in the end. Before this PR, it was not much better: every refactoring had to call rewrite on the appropriate rewriters in the end.

Moreover, every refactoring uses a different path through the interface methods to apply the refactoring.

So, without further cleanup or providing the session via method injection (see another comment above) I do not see how to put this in a base class.